include/swift/AST/Decl.h

```diff
@@ -4113,66 +4115,65 @@ class AbstractStorageDecl : public ValueDecl {
     /// return the address at which the object is stored.
     ComputedWithMutableAddress,
   };
+
+  static const size_t MaxNumAccessors = 255;
 private:
   AbstractStorageDecl *OverriddenDecl;
 
-  struct GetSetRecord;
-  
-  /// This is stored immediately before the GetSetRecord.
-  struct alignas(1 << 3) AddressorRecord {
-    AccessorDecl *Address = nullptr;        // User-defined address accessor
-    AccessorDecl *MutableAddress = nullptr; // User-defined mutableAddress accessor
-
-    GetSetRecord *getGetSet() {
-      // Relies on not-strictly-portable ABI layout assumptions.
-      return reinterpret_cast<GetSetRecord*>(this+1);
-    }
-  };
-  void configureAddressorRecord(AddressorRecord *record,
-                               AccessorDecl *addressor, AccessorDecl *mutableAddressor);
+  /// A record of the accessors for the declaration.
+  class alignas(1 << 3) AccessorRecord {
+    using AccessorIndex = uint8_t;
+    static const AccessorIndex InvalidIndex = 0;
 
-  struct alignas(1 << 3) GetSetRecord {
     SourceRange Braces;
-    AccessorDecl *Get = nullptr;       // User-defined getter
-    AccessorDecl *Set = nullptr;       // User-defined setter
-    AccessorDecl *MaterializeForSet = nullptr; // optional materializeForSet accessor
+    AccessorIndex NumAccessors;
+
+    /// The storage capacity of this record for accessors.  Always includes
+    /// enough space for adding opaque accessors to the record, which are the
+    /// only accessors that should ever be added retroactively; hence this
+    /// field is only here for the purposes of safety checks.
+    AccessorIndex AccessorsCapacity;
+
+    /// Either 0, meaning there is no registered accessor of the given kind,
+    /// or the index+1 of the accessor in the accessors array.
+    AccessorIndex AccessorIndices[NumAccessorKinds];
```

Comment:
Can you use llvm::TrailingObjects instead of a supposed fixed-sized array?

Reply:
That is a fixed-size array; it's indexed by AccessorKinds.  There *is* a trailing-objects use in that type, though.

Reply:
Oops, my brain misread NumAccessorKinds as MaxNumAccessors. Sorry for the noise!

Reply:
I've updated to use TrailingObjects for the part that's actually a trailing array.